libs/langchain-google-cloud-sql-pg/src/vectorStore.ts

```diff
@@ -401,6 +401,56 @@ export class PostgresVectorStore extends VectorStore {
 
     return docsList; 
   }
+
+  /**
+   * Create an index on the vector store table
+   * @param {BaseIndex} index 
+   * @param {string} name Optional
+   * @param {boolean} concurrently Optional
+   */
+  async applyVectorIndex(index: BaseIndex, name?: string, concurrently: boolean = false): Promise<void> {
+    if (index instanceof ExactNearestNeighbor) {
+      await this.dropVectorIndex();
+      return;
+    }
+
+    const filter = index.partialIndexes ? `WHERE (${index.partialIndexes})` : "";
+    const params = `WITH ${index.indexOptions()}`;
+    const funct = index.distanceStrategy.indexFunction;
+
+    if (!name) {
+      if (!index.name) {
+        index.name = this.tableName + DEFAULT_INDEX_NAME_SUFFIX;
+      }
+      name = index.name;
+    }
+
+    const stmt = `CREATE INDEX ${concurrently ? "CONCURRENTLY" : ""} ${name} ON "${this.schemaName}"."${this.tableName}" USING ${index.indexType} (${this.embeddingColumn} ${funct}) ${params} ${filter};`
+
+    if (concurrently) {
```

Comment:
This will work different than Python so we don't need to run the COMMIT or have the if else block